{
	"schema_version": "1.6.0",
	"id": "UBUNTU-CVE-2019-10856",
	"modified": "2025-01-13T10:21:55Z",
	"published": "2019-04-04T16:29:00Z",
	"related": [
		"CVE-2019-10856",
		"USN-5585-1"
	],
	"details": "In Jupyter Notebook before 5.7.8, an open redirect can occur via an empty netloc. This issue exists because of an incomplete fix for CVE-2019-10255.",
	"severity": [
		{
			"type": "CVSS_V3",
			"score": "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N"
		}
	],
	"affected": [
		{
			"package": {
				"ecosystem": "Ubuntu:18.04:LTS",
				"name": "jupyter-notebook",
				"purl": "pkg:deb/ubuntu/jupyter-notebook@5.2.2-1ubuntu0.1?arch=source&distro=bionic"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "5.2.2-1ubuntu0.1"
						}
					]
				}
			],
			"versions": [
				"4.2.3-4",
				"5.1.0-2",
				"5.2.1-2",
				"5.2.2-1"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "jupyter-notebook",
						"binary_version": "5.2.2-1ubuntu0.1"
					},
					{
						"binary_name": "python-notebook",
						"binary_version": "5.2.2-1ubuntu0.1"
					},
					{
						"binary_name": "python-notebook-doc",
						"binary_version": "5.2.2-1ubuntu0.1"
					},
					{
						"binary_name": "python3-notebook",
						"binary_version": "5.2.2-1ubuntu0.1"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2019/UBUNTU-CVE-2019-10856.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:20.04:LTS",
				"name": "jupyter-notebook",
				"purl": "pkg:deb/ubuntu/jupyter-notebook@6.0.3-2ubuntu0.1?arch=source&distro=focal"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "6.0.3-2ubuntu0.1"
						}
					]
				}
			],
			"versions": [
				"5.7.8-1",
				"6.0.0-1",
				"6.0.0-2",
				"6.0.2-1",
				"6.0.3-1",
				"6.0.3-2"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "jupyter-notebook",
						"binary_version": "6.0.3-2ubuntu0.1"
					},
					{
						"binary_name": "python-notebook-doc",
						"binary_version": "6.0.3-2ubuntu0.1"
					},
					{
						"binary_name": "python3-notebook",
						"binary_version": "6.0.3-2ubuntu0.1"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2019/UBUNTU-CVE-2019-10856.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:22.04:LTS",
				"name": "jupyter-notebook",
				"purl": "pkg:deb/ubuntu/jupyter-notebook@6.4.8-1ubuntu0.1?arch=source&distro=jammy"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "6.4.8-1ubuntu0.1"
						}
					]
				}
			],
			"versions": [
				"6.2.0-1",
				"6.4.5-2",
				"6.4.5-3",
				"6.4.5-4",
				"6.4.8-1"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "jupyter-notebook",
						"binary_version": "6.4.8-1ubuntu0.1"
					},
					{
						"binary_name": "python-notebook-doc",
						"binary_version": "6.4.8-1ubuntu0.1"
					},
					{
						"binary_name": "python3-notebook",
						"binary_version": "6.4.8-1ubuntu0.1"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2019/UBUNTU-CVE-2019-10856.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:24.10",
				"name": "jupyter-notebook",
				"purl": "pkg:deb/ubuntu/jupyter-notebook@6.4.13-2?arch=source&distro=oracular"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						}
					]
				}
			],
			"versions": [
				"6.4.12-2.2ubuntu1",
				"6.4.13-2"
			],
			"ecosystem_specific": {
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2019/UBUNTU-CVE-2019-10856.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:24.04:LTS",
				"name": "jupyter-notebook",
				"purl": "pkg:deb/ubuntu/jupyter-notebook@6.4.12-2.2ubuntu1?arch=source&distro=noble"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						}
					]
				}
			],
			"versions": [
				"6.4.12-2.2",
				"6.4.12-2.2ubuntu1"
			],
			"ecosystem_specific": {
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2019/UBUNTU-CVE-2019-10856.json"
			}
		}
	],
	"references": [
		{
			"type": "REPORT",
			"url": "https://ubuntu.com/security/CVE-2019-10856"
		},
		{
			"type": "REPORT",
			"url": "https://blog.jupyter.org/open-redirect-vulnerability-in-jupyter-jupyterhub-adf43583f1e4"
		},
		{
			"type": "REPORT",
			"url": "https://github.com/jupyter/notebook/commit/979e0bd15e794ceb00cc63737fcd5fd9addc4a99"
		},
		{
			"type": "REPORT",
			"url": "https://github.com/jupyter/notebook/compare/16cf97c...b8e30ea"
		},
		{
			"type": "REPORT",
			"url": "https://ubuntu.com/security/notices/USN-5585-1"
		},
		{
			"type": "REPORT",
			"url": "https://www.cve.org/CVERecord?id=CVE-2019-10856"
		}
	]
}
